nodejs/awsx/apigateway/api.ts

```diff
@@ -155,6 +155,32 @@ function isStaticRoute(route: Route): route is StaticRoute {
 export interface IntegrationRoute {
     path: string;
     target: pulumi.Input<IntegrationTarget> | IntegrationRouteTargetProvider;
+
+    /**
+    * Required Parameters to validate. If the request validator is set to ALL or PARAMS_ONLY, api
+    * gateway will validate these before sending traffic to the event handler. Parameter validation
+    * will get applied to all static resources defined by the localPath.
+    */
+    requiredParameters?: reqvalidation.Parameter[];
+
+    /**
+    * Request Validator specifies the validator to use at the method level. This will override anything
+    * defined at the API level.
+    */
+    requestValidator?: RequestValidator;
+
+    /**
+     * If true, an API key will be required for this route. The source for the API Key can be set at
+     * the API level and by default, the source will be the HEADER.
+     */
+    apiKeyRequired?: boolean;
+
+    /**
+     * Authorizers allows you to define Lambda authorizers be applied for authorization when the
+     * the route is called. The authorizer will get applied to all static resources defined by the
+     * localPath.
+     */
+    authorizers?: Authorizer[] | Authorizer;
```

Comment:
these four now seem common to most routes (except 'raw json') right?  If so, can we extract out to `interface BaseRoute` and then have the others extend from that so we don't have to copy this into all of them?

Reply:
Its not in the Raw Data Route. I'll spend some time thinking about how to add it to that..